elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -289,4 +291,28 @@ private SQLQuery toPageTotalSQL(SQLQuery sql) {
     public static String getClassAlias(Class<?> entityClass) {
         return getTypeAlias(entityClass);
     }
+
+    /**
+     * Functional interface for describing a method to supply JpaTransaction.
+     */
+    @FunctionalInterface
+    public interface TransactionCancel {
+        public void cancel(EntityManager entityManager);
+    }
+  
+    public class FutureImplementation implements Future<QueryResult> {
+        private final TransactionCancel transactionCancel;
+	private final EntityManagerFactory entityManagerFactory;
+        /**
+	 * Cancels transaction
+         */
+    	@Override
+	protected FutureImplementation(EntityManagerFactory entityManagerFactory, TransactionCancel transactionCancel) {
+	    this.transactionCancel = transactionCancel;
+	    this.entityManagerFactory = entityManagerFactory;
+	}
+    	public boolean cancel(boolean mayInterruptIfRunning) {
```

Comment:
You are missing a lot of the other Future functions.  There is more than just cancel.